etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/database/DatabasePartnerMetadataStorage.java

```diff
@@ -35,12 +45,25 @@ public static DatabasePartnerMetadataStorage getInstance() {
     public Optional<PartnerMetadata> readMetadata(final String uniqueId)
             throws PartnerMetadataException {
         try {
-            PartnerMetadata data = (PartnerMetadata) dao.fetchMetadata(uniqueId);
-            return Optional.ofNullable(data);
+            PartnerMetadata metadata =
+                    dao.fetchFirstData(
+                            connection -> {
+                                try {
+                                    PreparedStatement statement =
+                                            connection.prepareStatement(
```

Comment:
This is nice. For some time I've been thinking of having a wrapper class that will handle the prepared statements. Thoughts on having the wrapper class so that both storage classes can use it with the dao?